runtime/autoload/dist/ft.vim

```diff
@@ -590,23 +590,28 @@ export def FTprogress_cweb()
   endif
 enddef
 
+# These include the leading '%' sign
+var ft_swig_keywords = '^\s*%\%(addmethods\|apply\|beginfile\|clear\|constant\|define\|echo\|enddef\|endoffile\|extend\|feature\|fragment\|ignore\|import\|importfile\|include\|includefile\|inline\|insert\|keyword\|module\|name\|namewarn\|native\|newobject\|parms\|pragma\|rename\|template\|typedef\|typemap\|types\|varargs\|warn\)'
+# This is the start/end of a block that is copied literally to the processor file (C/C++)
+var ft_swig_verbatim_block_start = '^\s*%{'
+var ft_swig_verbatim_block_end = '^\s*}%'
+
 export def FTprogress_asm()
   if exists("g:filetype_i")
     exe "setf " .. g:filetype_i
     return
   endif
-  # This function checks for an assembly comment the first ten lines.
+  # This function checks for an assembly comment or a SWIG keyword or verbatim block in the first 50 lines.
```

Comment:
The idea here is that since a SWIG interface file is similar to a C/C++ file, it's not uncommon to have them start with a huge copyright notice or something, so parsing 10 lines isn't enough.